pom.xml

```diff
@@ -1733,7 +1733,7 @@
                             <!-- set heap size to work around https://github.com/travis-ci/travis-ci/issues/3396 -->
                             <argLine>
                                 ${jdk.surefire.argLine}
-                                -Xmx768m -Duser.language=en -Duser.country=US -Dfile.encoding=UTF-8
+                                -Xmx1280m -Duser.language=en -Duser.country=US -Dfile.encoding=UTF-8
```

Comment:
Is this needed because several tests are now going to use on-heap byte buffers?
Do those tests fail without this change?

Reply:
I don't know if this is needed. Right now I'm wrestling with inexplicable heap-usage errors early in the job cycle, like the following one. I thought this change might help but it doesn't. I need to do some more investigation into where this heap usage is coming from.

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.druid.guice.MetadataStorageTablesConfigTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.333 s <<< FAILURE! - in org.apache.druid.guice.MetadataStorageTablesConfigTest
[ERROR] org.apache.druid.guice.MetadataStorageTablesConfigTest.testSerdeMetadataStorageTablesConfig  Time elapsed: 7.076 s  <<< ERROR!
java.lang.OutOfMemoryError: GC overhead limit exceeded
	at java.util.zip.ZipCoder.getBytes(ZipCoder.java:80)
  	at java.util.zip.ZipFile.getEntry(ZipFile.java:316)
  	at java.util.jar.JarFile.getEntry(JarFile.java:240)
  	at java.util.jar.JarFile.getJarEntry(JarFile.java:223)
  	at sun.misc.URLClassPath$JarLoader.getResource(URLClassPath.java:1054)
  	at sun.misc.URLClassPath.getResource(URLClassPath.java:249)
  	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
  	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
  	at java.security.AccessController.doPrivileged(Native Method)
  	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
  	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
  	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  	at java.lang.ClassLoader.defineClass1(Native Method)
  	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
  	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
  	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
  	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
  	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
  	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
  	at java.security.AccessController.doPrivileged(Native Method)
  	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
  	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
  	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
  	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
  	at org.hibernate.validator.HibernateValidator.createGenericConfiguration(HibernateValidator.java:31)
  	at javax.validation.Validation$GenericBootstrapImpl.configure(Validation.java:276)
  	at javax.validation.Validation.buildDefaultValidatorFactory(Validation.java:110)
  	at org.apache.druid.guice.ConfigModule.configure(ConfigModule.java:39)
  	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
  	at org.apache.druid.guice.MetadataStorageTablesConfigTest$1.configure(MetadataStorageTablesConfigTest.java:48)
  	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340)
```

Reply:
I was able to reproduce OOMEs on processing tests locally. I think that the processing and merge buffers for groupBy tests need to stay off-heap, because there's a bunch of them and they're 10MB each. It adds up. I'll try organizing things so that they stay offheap but get freed properly.